src/library/scala/collection/immutable/Vector.scala

```diff
@@ -580,6 +580,31 @@ class VectorIterator[+A](_startIndex: Int, endIndex: Int)
 
     res
   }
+  override def copyToArray[B >: A](xs: Array[B], start: Int, len: Int): Int = {
```

Comment:
as display is not specialised will this work for copy to primitive types?

I think that Vector[Int] has display0 as Array[AnyRef], with casting & boxing

I think you will need to special case the AnyRef with the code below, and do super.copyToArray for the rest
@retronym is there a utility to copy with scala casting if the array types don't match?

Reply:
@mkeskells It does in fact work at least for `Int`, since that's what the benchmark was using

Reply:
@joshlemer this is not doing `copyToArray(Array[Int] ….)`
you are letting scala box and unbox.

```
  @Benchmark def concat(bh: Blackhole): Any = {
    bh.consume(first ++ second)
  }

```

Reply:
Aren't the elements boxed already because they are in Vector's `Array[AnyRef]`?

Reply:
Yes, `Vector[Int]` contains boxes.

Reply:
but to parameter to copyToArray could be a primitive array.